lib/state_change_syncer.go

```diff
@@ -174,8 +175,12 @@ func (stateChangeEntry *StateChangeEntry) RawDecodeWithoutMetadata(blockHeight u
 	ancestralRecord := stateChangeEntry.EncoderType.New()
 	if exist, err := DecodeFromBytes(ancestralRecord, rr); exist && err == nil {
 		stateChangeEntry.AncestralRecord = ancestralRecord
+		stateChangeEntry.AncestralRecordBytes = EncodeToBytes(blockHeight, ancestralRecord)
```

Comment:
I don't think this is problematic, but it could slow things down a lot. Encode/Decode is one of the biggest consumers of sync time I think. Consider skipping the encode/decode if you can. But it's also fine to leave it if it's hard to get rid of it.

Reply:
This is a good call-out. There are instances in the state consumer where we do use the raw bytes on the decoded state change entry ancestral record, and we haven't noticed any noticeable slowdowns as a result of the encoders specifically, but in the future if we do find that these become a bottleneck, we could consider modifying our consumer logic to only decode when we need to, e.g. a subsequent encode call on `stateChangeEntry.AncestralRecordBytes`.